dev/fattest.simplicity/src/componenttest/rules/repeater/MicroProfileActions.java

```diff
@@ -368,7 +388,46 @@ public static RepeatTests repeat(String server, FeatureSet firstFeatureSet, Feat
      * @return                          A RepeatTests instance
      */
     public static RepeatTests repeat(String server, TestMode otherFeatureSetsTestMode, FeatureSet firstFeatureSet, FeatureSet... otherFeatureSets) {
-        return repeat(server, otherFeatureSetsTestMode, ALL, firstFeatureSet, otherFeatureSets);
+        Set<FeatureSet> otherFeatureSetsSet = new HashSet<>(Arrays.asList(otherFeatureSets));
+        return repeat(server, otherFeatureSetsTestMode, ALL, firstFeatureSet, otherFeatureSetsSet);
+    }
+
+    /**
+     * As {@link RepeatActions#repeat(String, TestMode, Set, FeatureSet, Set)} except that if {@code firstFeatureSet} isn't compatible with the current Java version, we try to
+     * replace it with the newest set from {@code otherFeatureSets} that is compatible.
+     *
+     * @param  server                   The server to repeat on
+     * @param  otherFeatureSetsTestMode The test mode to run the otherFeatureSets
+     * @param  allFeatureSets           All known FeatureSets. The features not in the current FeatureSet are removed from the repeat
+     * @param  firstFeatureSet          The first FeatureSet to repeat with. This is run in LITE mode.
+     * @param  otherFeatureSets         The other FeatureSets to repeat with. These are in the mode specified by otherFeatureSetsTestMode
+     * @return                          A RepeatTests instance
+     */
+    private static RepeatTests repeat(String server, TestMode otherFeatureSetsTestMode, Set<FeatureSet> allFeatureSets, FeatureSet firstFeatureSet,
+                                      Set<FeatureSet> otherFeatureSets) {
+
+        // If the firstFeatureSet requires a Java level higher than the one we're running, try to find a suitable replacement so we don't end up not running the test at all in LITE mode
+        int currentJavaLevel = JavaInfo.forCurrentVM().majorVersion();
+        if (currentJavaLevel < firstFeatureSet.getEEVersion().getMinJavaLevel()) {
+
+            List<FeatureSet> allSetsList = new ArrayList<>(Arrays.asList(ALL_SETS_ARRAY));
+            Collections.reverse(allSetsList); // Reverse list so newest MP version is first in list
+
+            Set<FeatureSet> candidateFeatureSets = otherFeatureSets;
+
+            // Find the newest MP feature set that's in otherFeatureSets and is compatible with the current java version
+            Optional<FeatureSet> newestSupportedSet = allSetsList.stream()
+                            .filter(s -> candidateFeatureSets.contains(s))
+                            .filter(s -> s.getEEVersion().getMinJavaLevel() <= currentJavaLevel)
+                            .findFirst();
+
+            if (newestSupportedSet.isPresent()) {
+                firstFeatureSet = newestSupportedSet.get();
+                otherFeatureSets = new HashSet<>(otherFeatureSets);
```

Comment:
This will work until we have a MP 6.1 or 7.0 right?  Don't we need to remove the ones that will not work with the current JDK from the otherFeatureSets?

Reply:
It could be done in the RepeatActions.repeat method that if the MP level is not supported on the current JDK in the other featureSets, we just do do the `andWith` call for that MP level.

Reply:
I thought the problem here was just that if a feature set wasn't supported on the current JDK, it would be skipped because `withMinJavaLevel()` was set on the repeat action.

The only problem then was when the first feature set gets skipped because that means we don't run any tests at all in lite mode.

Reply:
Oh yeah.  It was late.  :)